workflow/snakemake_rules/download_from_fauna.smk

```diff
@@ -0,0 +1,129 @@
+
+path_to_fauna = '../fauna'
+localrules: download_sequences, download_titers, parse, metadata
+#
+# Define titer data sets to be used.
+#
+def _get_tdb_databases(wildcards):
+    if wildcards.center in ['cdc', 'crick', 'niid', 'vidrl']:
+        return wildcards.center + "_tdb tdb"
+    else:
+        return "cdc_tdb crick_tdb niid_tdb vidrl_tdb tdb"
+
+
+def _get_tdb_assays(wildcards):
+    if wildcards.assay == 'fra':
+        return 'fra,hint'
+    if wildcards.assay == 'hi':
+        return 'hi,hi_oseltamivir'
+    return wildcards.assay
+
+rule download_sequences:
+    message: "Downloading sequences from fauna"
+    output:
+        sequences = "data/{lineage}/raw_{segment}.fasta"
+    params:
+        fasta_fields = " ".join(fasta_fields)
+    conda: "environment.yaml"
+    shell:
+        """
+        python3 {path_to_fauna}/vdb/download.py \
+            --database vdb \
+            --virus flu \
+            --fasta_fields {params.fasta_fields} \
+            --resolve_method split_passage \
+            --select locus:{wildcards.segment} lineage:seasonal_{wildcards.lineage} \
+            --path data \
+            --fstem {wildcards.lineage}/raw_{wildcards.segment}
+        """
+
+rule download_titers:
+    message: "Downloading titers from fauna: {wildcards.lineage}, {wildcards.assay}, {wildcards.center}"
+    output:
+        titers = "data/{lineage}/{center}_{passage}_{assay}_titers.tsv"
+    params:
+        dbs = _get_tdb_databases,
+        assays = _get_tdb_assays
+    conda: "environment.yaml"
+    shell:
+        """
+        python3 {path_to_fauna}/tdb/download.py \
+            --database {params.dbs} \
+            --virus flu \
+            --subtype {wildcards.lineage} \
+            --select assay_type:{params.assays} serum_passage_category:{wildcards.passage} \
+            --path data \
+            --fstem {wildcards.lineage}/{wildcards.center}_{wildcards.passage}_{wildcards.assay}
+        """
+
+rule parse:
+    message: "Parsing fasta into sequences and metadata"
+    input:
+        sequences = rules.download_sequences.output.sequences
+    output:
+        sequences = "data/{lineage}/{segment}.fasta",
+        metadata = "data/{lineage}/metadata_{segment}.tsv"
+    params:
+        fasta_fields =  " ".join(output_fasta_fields),
+        prettify_fields = " ".join(prettify_fields)
+    conda: "environment.yaml"
+    shell:
+        """
+        augur parse \
+            --sequences {input.sequences} \
+            --output-sequences {output.sequences} \
+            --output-metadata {output.metadata} \
+            --fields {params.fasta_fields} \
+            --prettify-fields {params.prettify_fields} && \
+        rm {input}
+        """
+
+
+rule metadata:
+    input:
+        segment_metadata = lambda w: [f"data/{w.lineage}/metadata_{segment}.tsv" for segment in config['segments']]
+    output:
+        "data/{lineage}/metadata.tsv"
+    params:
+        segments = lambda w: config['segments']
```

Comment:
After working with this workflow more, I'm not sure we want a single metadata file per lineage. This approach works when we don’t have any segment-specific metadata, but our work with SARS-CoV-2 suggests we will want this kind of metadata. Some example columns we would want to include are:

* GISAID segment id (the id of the segment sequence, different from the id of the virus isolate)
* GenBank segment id (the accession of the sequence in GenBank)
* Nextclade QC metrics for each sequence (length, number of gaps, runs of SNVs, rare mutations, etc.)
* GISAID- or Nextclade-annotated sequence-specific metadata including amino acid substitutions, nucleotide mutations, insertions, and deletions

These segment-specific columns will get unwieldy especially when we include all segments and end up with columns like `aaSubstitutions_ha`, `aaSubstitutions_na`, `aaSubstitutions_mp`, etc. for every combination of segment and segment-specific column. I think we should keep the pattern of one metadata file to one sequence file with the segment-specific data in the standard "long" format and include a "segment" column that we can query on. We can handle the set logic of which strains have the segments we want using other tools.

Reply:
Ok, after posting that comment, I almost immediately ran into the issue that probably motivated the single metadata file; augur filter doesn't work with multiple segment inputs, so we can't subsample the same strains across segments unless we first collect the metadata together into a single file.

This potential multi-segment functionality of augur filter (or joining of multiple metadata files by a key) relates to and potentially conflicts with our desire to support multiple metadata inputs for nonsegmented genomes (or concatenating of multiple metadata files). I'll work on moving the code from the `run` block that handles this logic in this PR into its own function, but maybe the better long-term solution is to support this kind of multi-segment joining as a separate augur command that could be piped into augur filter, instead of expecting augur filter to know about multiple segments.